src/python/pants/engine/README.md

```diff
@@ -17,63 +17,69 @@ Once the engine is instantiated with a valid set of `@rule`s, a caller can synch
 computation of any of the product types provided by those `@rule`s by calling:
 
 ```python
-# Request a ThingINeed (a `Product`) for the thing_i_have (a `Subject`).
+# Request a ThingINeed (a `Product`) for a thing_i_have (a `Param`).
 thing_i_need, = scheduler.product_request(ThingINeed, [thing_i_have])
 ```
 
 The engine then takes care of concurrently executing all dependencies of the matched `@rule`s to
 produce the requested value.
 
-### Products and Subjects
+### Products and Params
 
 The engine executes your `@rule`s in order to (recursively) compute a `Product` of the requested
-type for a given `Subject`. This recursive type search leads to a very loosely coupled (and yet
+type for a set of `Param`s. This recursive type search leads to a loosely coupled (and yet
 still statically checked) form of dependency injection.
 
-When an `@rule` runs, it runs for a particular `Subject` value, which is part of the unique
-identity for that instance of the `@rule`. An `@rule` can request dependencies for different
-`Subject` values as it runs (see the section on `Get` requests below). Because the subject for
-an `@rule` is chosen by callers, a `Subject` can be of any (hashable) type that a user might want
-to compute a product for.
+When an `@rule` runs, the set of `Param`s that it transitively consumes make up the unique identity
+for that instance of the `@rule`. Any hashable type with useful equality may be used as a `Param`,
+and additional `Params` can be provided to an `@rule`'s transitive dependencies via `Get`
```

Comment:
Right now I mostly read it as "at rule", in order to disambiguate from other things called rules (the word is fairly overloaded). Will leave for now, since I don't think this is the only place we do this.

Reply:
Okay that's fine with me. Threw me off at first, but now I'm used to it. 

I imagine this is something that's half of people would read as `an` and half as `a`, so there's no way to please everyone.